karl/textindex.py

```diff
@@ -84,3 +94,186 @@ def max_ranked(self):
             max_ranked = int(get_setting(self, 'pgtextindex.max_ranked', 6000))
             self._v_max_ranked = max_ranked
         return max_ranked
+
+    def drop_and_create(self):
+        cm = self.connection_manager
+        conn = cm.connection
+        cursor = cm.cursor
+        try:
+            # Create the table.
+            stmt = """
+            DROP TABLE IF EXISTS %(table)s;
+
+            CREATE TABLE %(table)s (
+                docid INTEGER NOT NULL PRIMARY KEY,
```

Comment:
To *exactly* match Python's ``int`` (and therefore also ZODB's keys, from my understanding) it should be ``bigint`` instead of ``integer``, even if hitting the bounds seems rather unlikely. 

Also see http://www.postgresql.org/docs/9.1/static/datatype-numeric.html.